New sprites for exploding barrel, light column and armor bonus

shinobody opened the pull request:
- 3 new barrel sprites (added red, warning sign more visible, redesigned into propane tank)
- new light column that looks like an actual lamp
- 2 new armor bonus sprites that look like shield/armor scales

AXDOOMER commented:
On bexpa0.png, there are two empty columns on each side of the sprite. Maybe remove them and realign the sprite's offset.

shinobody commented:
Oh, right, for some reason I assumed the sprite has to be dimensions of the original, idk why. Fixed.

shinobody commented:
Wait uh... I fixed the columns, but how do I realign the sprite's offset?

Talon1024 commented:
You have to set the offsets in buildcfg.txt. The numbers beside each sprite name in buildcfg.txt are the offsets.

shinobody commented:
Okay done!

Talon1024 commented:
BTW, where are the BAR1A0/BAR1B0 sprites? Without those, the explosive barrel will still look like a barrel, not a propane tank.

shinobody commented:
Oh crap, I'm so sorry, I'm new to this, Please bear with me. Fixed, and changed offsets for those new graphics. I'm really sorry, I will remember to do all that next time.

axredneck commented:
For me they look more like robots than propane tanks

shinobody commented:
I tried to follow the reference as best as i could - [https://imgur.com/a/mIzFx0u]
Also the fact that they look like propane tanks is secondary (it's fluff), the most important part is that the warning sign is vibrant and saturated, and there's some extra red on it, making it clear that it is, indeed, and explosive barrel (most explosive barrels in games released after Doom are red, so this should be obvious to new people playing Freedoom). Currently the warning sign is so faded that you can easily think they're just decoration (I did that on my first blind playtrhough).

If you have critque regarding any specific part of the sprite, I'm more than happy to listen and make edits.

axredneck commented:
> the warning sign is vibrant and saturated, and there's some extra red on it, making it clear that it is, indeed, and explosive barrel

for me it looks like broken heart

shinobody commented:
It's literally the same sign as before, just recolored?

mc776 commented:
Not a fan of the logic around making the warning sign red. Most folks playing Freedoom will know what the green barrels are for, and taking away from the green will detract from *that* well-established visual language.

Can you make them more cylindrical like the originals? These just don't have that same sense of weight to them - this I think is one of those times where the resulting silhouette for ease of recognition should take precedence over correspondence to IRL symbols.

That all said, I do like the carrying handles.

shinobody commented:
The barrels we have currently aren't green, they're just vaguely grayish and slightly olive, they look like decoration. Again, I played it blind and had no clue that they were meant to be barrels until a stray shotgun pellet accidentally detonated one. We can't discount that kind of stuff *unless* the plan is to create actual blind playtests for new builds (which would be great to be honest), and I'm not aware of any such plans. Blind playtesting is key for making actually good games. And for the record warning sign in current version is ALREADY red and yellow, it's just faded red and yellow. You might be thinking of vanilla Doom barrels (which are indeed green), but current version looks nothing like them.
Here's comparison for clarity - https://imgur.com/a/4mjYRHZ - left is new, right is old. Old barrels are not really green, don't look much like a barrel or honestly, like anything in particular.

In addition, Doom was released in 1994 when explosive barrels were a new concept and they could've made them look however they wanted. We're developing this game now, for audience that might have never played Doom (if they had access to a Doom wad, why wouldn't they just use that?) - Freedoom is freely available in several app stores, like Goole Play Store and Ubuntu Software Center. We're not just developing for existing Doom nerds, we have to think of new player experience. And the trope in modern game design says, exploding barrels are red (as seen in Half Life 2, Resident Evil 4 and even C&C RA3). I'm just following it.

Also I'm not sure why barrels _need_ a sense of weight - they're not primarily physics objects. Primary purpose of game graphics is to convey important informations (second is to look pretty), in this case - whether they're dangerous/explosive/interactable is much more important than their sense of weight. We need to prioritize clarity and affordance.

As for filesize, I imagine that might be due to differences in PNG compression or in palette indexing, I doubt anything will be different when they're actually converted to Doom graphics.

axredneck commented:
> The armour bonuses are an unmitigated improvement, for the same reasons as my above criticisms about the barrels.

But i like these new armor scales

mc776 commented:
The last thing this project needs is more same-same conformity to gamer tropes.

Sorry, but these new sprites are just uglier than the originals on every level.

And yes, that *is* green on the current ones.

shinobody commented:
One, "uglier" is not really constructive critique I can do anything with, and you're so far the first person to say they're ugly, other people offering feedback in the Discord server said it looks okay. Like, what makes them ugly? Texture? Color? Shading? Shape?
I can remodel the dimensions so they're more top-heavy, sure, that's actually good feedback - if you're want me to actually implement your comments, keep them more like that.
And I mean, I *could* try recoloring the warning sign to be green instead? Maybe that could still work? I could also implement axredneck's feedback and separate the exclamation mark from the sign border so it doesn't look like a broken heart.

Honestly I'd kill for a blind playtest so we could actually compare how good a group of blind players is at spotting if those are barrels. I know I couldn't, but it was just me being silly? Is this a real problem? We have no clue, and blind playtesting and iteration are crucial in game development.
Or at least idk, as a substitute a general poll running for 2 weeks on Doomworld forums. Also perhaps some kind of dedicated "art director" role (someone experienced in making game graphics and for a long time on the project) who would have last word on approving visual assets would be nice, too.

But I can't help but comment on "_The last thing this project needs is more same-same conformity to generic gamer tropes_ " Do you mean like... **conformity to good game design?** Did you seriously just call Half Life 2 "conforming to generic gamer tropes"? I'm sorry but HL2 - which easily helped popularize red explosive barrels as a key gameplay element - is currently a much better game than Freedoom and I don't think there are many people who would disagree. I say this and I **like** Freedoom (I wouldn't be here if I didn't). And that's partially because everything is designed with clarity of purpose, with affordance in mind. SMG looks like an SMG, explosive barrel looks like a barrel, gunship looks like a gunship, Combine machinegun looks like a machinegun - and this is all despite all of them being drawn in a dystopian sci-fi theme consistent with the rest of the game.
Like, we're not creating some heady unique cerebral high-art game like Pathologic or That Dragon Cancer here, the Freedoom is about as much of a literal Doom clone as it's physically possible. Our goal is to simply make it as easy to learn, as hard to master and as fun - as possible. Red barrels are probably an obvious choice, part of it, for that purpose.

Here's some videos on why explosive barrels are red that might change your mind, it's a type of 'affordance' - design that informs you of purpose of an object through audiovisual design:
https://www.youtube.com/watch?v=3WeVAp7JQGk
https://www.youtube.com/watch?v=QCSXEKHL6fc
https://www.youtube.com/watch?v=LMYEE8lvlvA

Also I guess it's current barrels are technically lightly green if you consider olive as a shade of green (technically this olive is yellow-green in hue - proof with a screencap of color wheel, it's right on the border between yellow and green - https://imgur.com/a/501O6Gn), but it doesn't use the same bright green color ramp as barrels from OG Doom. So it's pointless anyway. And they also don't look like barrels - I thought they looked kinda like propane tanks so that's the direction I went in. I *could* try to make them just grey barrels of glowing green goop like in Doom, but I thought the point of this project was to *not* just copying Doom and try to be original from it when possible?

So yeah, I can implement **some** of this feedback - making them more green, more top heavy, making the sign look less like a broken heart - but 'uglier' is not constructive critique, that's just an opinion.

BTW, Is it possible to move this discussion to the Freedoom discord? It's been 2 days since my proposal and I imagine this would go a lot faster if we could discuss it at a pace higher than 1 post per day. Since then I finished some other contributions that I'd like to submit (edited version of E1M1), but I pretty much have to wait until this gets approved or rejected unless I want to do some complex stuff.

shinobody commented:
Anyway, propane tank wasn't my idea. I thought the old one looks cool, people in the server said that it needs replacement.
How about this as a new version? More green, looks basically identical to old version, while the warning sign is much more pronounced and draws player's attention - and doesn't look like a broken heart.
https://imgur.com/a/xNQPCb0

Xindage commented:
this one remenber the one that i knew, maybe i could accept this one better, and about it needing a replacement that was because maybe you've heard of few saying about it, not all, I was against changing it but well i  dunno why i haven't said anything.

shinobody commented:
Well, I'm new so I'm not sure who specifically I should listen to. I noticed you're a mod, but I'm not sure about others.

Anyway, if that's acceptable, I could make sprites for this version and submit them.

Speaking of, is it legal to take the rejected propane tank sprites and put them on OpenGameArt as CC BY 3.0? Someone in the server suggested this.

Xindage commented:
yes i'm a moderator but that wont mean that you have to consider my voice higher than them because it, i'm just a contributor like them. I was just pointing that i still like the classic barrel, it's basic, and don't call too much attention, if you consider the doom barrel it's not so bright too, the both are dangerous but normal part of the ambient/map, they dont need to be so visible.
Doom teach you how they're dangerous by giving lot of them early in game to you play with them, after a time you get used to them and their appearance.

shinobody commented:
I mean, I see no reason why we can't do both. I'm about to submit a fixed version of E1M1, I could include a textless tutorial section on the barrels, in addition to the new sprites.

shinobody commented:
I could also make the glowing exclamation mark subdued if that would help.

mc776 commented:
Are you seriously arguing about whether [olive green](https://en.wikipedia.org/wiki/Olive_(color)) is green?

I know quite a few people in this project don't check the Discord, myself included (with one recent exception when I tried to look at the discussion around this and found the video of the Octaminator revenant in action instead).

On a purely subjective but perhaps more convincing level: the propane-tank-like design doesn't do for me what it sets out to do. It's just another decorative container thingy, no more or less than the current barrel. The thing about recognizing the barrels as the gameplay element they are is that there's so much stuff that's red in the game already that it'll just get lost.

Here's the problem I see in the problem you see yourself trying to fix with this change: **there is no way to design the sprite to make it *inherently* obvious to someone who's never played Doom before that if you shoot this a damaging explosion will happen.** As a gameplay mechanic it's *completely unique within the game,* nothing else reacts to damage like this - the **only** way for a totally new player to discover this is by trial and error. The only unequivocal symbol I can think of is a wooden powder keg clearly marked as such, but that makes no sense in this setting.

As for why I'm pushing back against this so much: a lot of custom maps (including those for Freedoom - heck, even Freedoom's own maps) have things colour-coordinated and suddenly swapping the barrels from grey-with-green to a design with predominantly warm colours is going to mess with a lot of custom maps' aesthetics. The loss of the diegetic association with the green nukage would also break a lot of the thematic consistency within those maps; making them into the obvious all-red cliché-barrel would turn these places into a gaudy Yuletide nightmare. 🎄🎅

The newer version with the blinking warning sign is vastly better, though, and does give some indication that it's capable of some kind of interactivity if not necessarily that it's a shootable environmental hazard.

shinobody commented:
_"Here's the problem I see in the problem you see yourself trying to fix with this change: there is no way to design the sprite to make it inherently obvious to someone who's never played Doom before that if you shoot this a damaging explosion will happen. As a gameplay mechanic it's completely unique within the game, nothing else reacts to damage like this - the only way for a totally new player to discover this is by trial and error. The only unequivocal symbol I can think of is a wooden powder keg clearly marked as such, but that makes no sense in this setting."_ Then please explain me how hundreds of games manage just that? They do that by making them fuel cans (like HL2) or propane tanks (like Uncharted) that are red in color. Which is what I tried here. You're trying to design this game in a vacuum but we're not in a vacuum, hundreds of games have been released since 1994, and hundreds before that.
You're claiming that this problem can't be solved when it was solved long ago. Did you watch the videos I linked? Seriously, they might change your mind.

For a matter of fact, I think background decoration should not be bright red anyway and that's another problem to resolve, further on the list. That's why I submitted new light column, it looks interactive when it isn't. Another red decoration, current Evil Eye, is also on the chopping block - partially because it will be almost entirely incompatible with Sigil, which uses Evil Eye as a gun switch target, while our current sprite doesn't look like a target - kat and I and some others (like HorrorMovieRei) have some ideas for a new one that will look similar to vanilla, while fitting a sci-fi theme (current dominant idea seems to be a hologram projector).

I agreed that you can see olive green as green. Technically from perspective of color theory that hue is called yellow-green or chartreuse - https://www.color-meanings.com/color-wheel-theory-complementary-colors/  - which was the reason for our disagreement, we see olive green in different frameworks. Considering that I'm trying to reconcile on this point, which is pretty much meaningless anyway, I wish we could drop it.

And trust me, I very much understand the problem of color coordination. I addressed it several time in the Discord, I could've sworn I addressed it here too - I guess I didn't. Sorry.

So how about this: it seems both mc776 and Xindage prefer the blinking brighter warning sign on a current version of the sprite. Is there any tweaks I need to do to this version, so I can commit it? I imagine if we need to, I could change the warning sign to be bright green instead of red (because vanilla Doom barrel uses bright green) - here's a sample: https://imgur.com/a/tISv6NV
Which one do y'all like the most? 1, 2 3 or 4? I think 1 (white bg) looks the best, and it also has pretty much the same color scheme as vanilla barrel, green and gray.

Xindage commented:
2 yellow background.

shinobody commented:
Added new barrel to the roster, see the imgur again, number 5. Xindage signed off on it in the Discord.

shinobody commented:
Okay, after two hours of intense debate on Discord, here's a new barrel - mostly by A Kat aka Korp, but also by MissLavender and by me (Shino1).
https://imgur.com/a/KukRAvW
I think it's optimal.

MissLav commented:
:) good stuff

mc776 commented:
That last one is definitely an improvement.

Calinou reviewed (approved):
All 3 sprites look better than the old ones to me, good work :slightly_smiling_face:

shinobody commented:
mc776, I'm glad we were able to reach consensus. Honestly, this makes me really happy.
And thanks to everyone for feedback!

shinobody commented:
Okay, fixed the armor bonus sprites. I guess I didn't commit every file I made. Sorry, I said I'm new to this.
